src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/DefaultArmorColour.java

```diff
@@ -21,42 +21,62 @@
 
 import com.google.gson.JsonObject;
 import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
-import io.github.moulberry.notenoughupdates.util.Utils;
+import io.github.moulberry.notenoughupdates.autosubscribe.NEUAutoSubscribe;
 import net.minecraft.item.ItemArmor;
 import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.JsonToNBT;
+import net.minecraft.nbt.NBTException;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.event.world.WorldEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 
+import java.util.HashMap;
+import java.util.Map;
+
+@NEUAutoSubscribe
 public class DefaultArmorColour {
 
-	private static boolean shownError = false;
+	private static Map<String, Integer> armorCache = new HashMap<>();
```

Comment:
Consider using a `Map<String, Integer> armorColorCache = new IdentityHashMap<>();` instead, so you can cache the item resolution query step as well.